<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Example- Ensembl - API-CARD</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href='https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Example- Ensembl";
    var mkdocs_page_input_path = "multiomics-Ensembl.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> API-CARD</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="tocbase current">
    
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="..">Home</a>
  </li>
    
      
  <li class="navtree toctree-l1 label">
    <p class="caption">API Card</p>
  </li>


  

  
    <li class="navtree toctree-l1 group">
      <ul class="navtree subnav-l1 current">
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../nci-genomic-data-commons/">NCI Genomic Data Common</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../ucsc-genome-browser/">USCS Genome Browser</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../ensembl-rest-api/">Ensebl REST API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../new-gene-info/">New Gene Info</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../gene-ontology/">Gene Ontology</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../gwasrappid-catalog/">GWAS Rappid Catalog</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../omim-api/">OMIM API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../web-gestalt-web-api/">WebGesalt Web API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../cbio-portal/">cBio Portal</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../scanty/">Scanpy</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../pybio-mart/">PyBio Mart</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../pubtator-central-api/">PubTator Central API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../disgenet/">Disgenet API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../kegg/">KEGG API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../mgnify/">Mgnify API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../scop/">SCOP API</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../uniprot/">Uniprot API</a>
  </li>
        
      </ul>
    </li>
    
      
  <li class="navtree toctree-l1 label">
    <p class="caption">Multiomics Integration</p>
  </li>


  

  
    <li class="navtree toctree-l1 group">
      <ul class="navtree subnav-l1 current">
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-Scanpy-PyBio-Mart/">Example- Scanpy and  PyBioMart</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-Scanpy-GO-Uniprot/">Example- Scanpy, GO and Uniprot</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-Scanpy-GO/">Example- Scanpy Go</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-PyBiomart-Scanpy-GO-Uniprot/">Example- Scanpy, PyBioMart, ScanPy, Go, and Uniprot</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-omim-ucsc-go-Uniprot-DrugBank/">Example- Omim, USGC, GO, Uniprot and DrugBank</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-omim-ucsc/">Example- Omim and UCSC</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-NCI-GDC-Scanpy-WebGestalt/">Example- NCI, GDC ScanPy and EebGestalt</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-NCI-GDC-GO/">Example- NCI, GDC and GO</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-MyGeneInfo-cBioPortal/">Example- MyGenInfo and cBioportal</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-GWAS-WebGestalt/">Example- GWAS WebGestalt</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../multiomics-GWAS-eQTLs/">Example- GWAS eQTLS</a>
  </li>
        
          


  
    
    <li class="navtree toctree-l2 page current">
      <a class="current" href="./">
        Example- Ensembl
          <span class="toctree-expand"></span>
      </a>
    </li>
    
      

  <li class="toctree-l2 current with-children">
    <a href="#api-based-method-retrieves-variant-consequence-proportions-for-human-diseases">
      API-based method retrieves variant consequence proportions for human diseases
      <span class="toctree-expand"></span>
    </a>
  </li>



  <li class="toctree-l2 current">
    <ul class="subnav-l2 current">
    
      
        <li class="toctree-l3">
          <a class="toctree-l4" href="#explored-by">Explored by:</a>
        </li>
    
      
          

  <li class="toctree-l3">
    <a href="#description">
      Description
      <span class="toctree-expand"></span>
    </a>
  </li>



  <li class="toctree-l3">
    <ul class="subnav-l3 toc-hidden">
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#project-overview">Project overview</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#function-to-access-ensembl-rest-api-features">Function to access Ensembl Rest API features</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#function-to-find-gene-regions-in-our-list-of-dictionaries">Function to find gene regions in our list of dictionaries</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#function-to-return-all-variants-associated-with-regions">Function to return all variants associated with regions</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#use-vep-to-get-variant-consequence">Use VEP to get variant consequence</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#calculate-proportions-of-consequence-in-a-variant-list">Calculate proportions of consequence in a variant list</a>
        </li>
    
      
        <li class="toctree-l4">
          <a class="toctree-l5" href="#results-proportion-of-variant-consequences-in-alzheimers-disease">Results: Proportion of variant consequences in Alzheimer’s disease</a>
        </li>
    
    </ul>
  </li>

      
    
    </ul>
  </li>


  
        
      </ul>
    </li>
    
  </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">API-CARD</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Multiomics Integration &raquo;</li>
        
      
    
    <li>Example- Ensembl</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h2 id="api-based-method-retrieves-variant-consequence-proportions-for-human-diseases">API-based method retrieves variant consequence proportions for human diseases</h2>
<h4 id="explored-by">Explored by:</h4>
<p><code>DARCI-MAHER, NICHOLAS WAXTER</code>, <code>RAZMA, CONNOR JOHN</code>- <code>(Bioinfo 201 - Winter 2022)</code></p>
<h3 id="description">Description</h3>
<h4 id="project-overview">Project overview</h4>
<ol>
<li>
<p>Assume we are given a disease of interest</p>
</li>
<li>
<p>Use the Ensembl phenotype annotation endpoint to extract genomic
regions associated with the disease</p>
</li>
<li>Use the Ensembl overlap endpoint to find variants in these regions 4.
Use the Ensembl Variant Effect Predictor (VEP) endpoint to discover the
consequences of each variant</li>
<li>Calculate the proportions of each consequence for the disease</li>
</ol>
<h4 id="function-to-access-ensembl-rest-api-features">Function to access Ensembl Rest API features</h4>
<p>● Input: Extension specifying application
● Output: Object associated with application
● In our case we will use this function to get a list of dictionaries with areas of the genome associated with alzheimer's</p>
<p><img alt="img" src="../img/ensemble1.png" /></p>
<h4 id="function-to-find-gene-regions-in-our-list-of-dictionaries">Function to find gene regions in our list of dictionaries</h4>
<p>● Input: List of dictionaries of areas of genome associated with a disease 
● Output: List of gene regions that are associated with a disease</p>
<p><img alt="img" src="../img/ensemble2.png" /></p>
<h4 id="function-to-return-all-variants-associated-with-regions">Function to return all variants associated with regions</h4>
<p>● Input: A list of regions associated with a disease 
● Output: All variants associated with those regions</p>
<p><img alt="img" src="../img/ensemble3.png" /></p>
<h4 id="use-vep-to-get-variant-consequence">Use VEP to get variant consequence</h4>
<p>● Input: a single variant ID (rsID) 
● Output: The most severe consequence of that variant (e.g. missense, splice  site, stop gained, etc.)</p>
<p><img alt="img" src="../img/ensemble4.png" /></p>
<h4 id="calculate-proportions-of-consequence-in-a-variant-list">Calculate proportions of consequence in a variant list</h4>
<p>● Input: a list of variant IDs 
● Output: proportion of consequences for the disease associated with the list  of variants</p>
<p><img alt="img" src="../img/ensemble5.png" /></p>
<h4 id="results-proportion-of-variant-consequences-in-alzheimers-disease">Results: Proportion of variant consequences in Alzheimer’s disease</h4>
<p>● Ran our full pipeline on Alzheimer’s  disease (EFO ID=0000249) 
● Found that the most common<br />
consequences in Alzheimer’s variants are: 
1. Intron variant (35%) 
2. Missense variant (14%) 
3.  3’ UTR variant (13%)</p>
<p><img alt="img" src="../img/ensemble6.png" /></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="../multiomics-GWAS-eQTLs/" class="btn btn-neutral" title="Example- GWAS eQTLS"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../multiomics-GWAS-eQTLs/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
